### DISCUSSION TITLE
Issue with SOAP Descriptor: AttributeError: module 'dscribe' has no attribute 'ext'

### DISCUSSION THREAD

**Discussion by deepakpawar2310** [SINGROUP/dscribe#118] [General]
Hi Everyone !
 I am currently using SOAP (Smooth Overlap of Atomic Positions) descriptors for molecular systems. However, I am facing an issue while trying to create a SOAP descriptor using the dscribe library.

The specific error message I encounter is:

**AttributeError: module 'dscribe' has no attribute 'ext'**

Upon researching this error, it seems that in the version of dscribe I am using, the ext attribute is no longer available. This has caused problems with the creation of the SOAP descriptor in my code.

Here's a snippet of the relevant code:

**from dscribe.descriptors import SOAP

# SOAP descriptor settings
species = ["Cu"]
cutoff = r_cut
n_max = n_max
l_max = l_max
periodic = False
sigma = 0.01 * cutoff

# Create the SOAP descriptor
soap_descriptor = SOAP(
    species=species,
    rcut=cutoff,
    nmax=n_max,
    lmax=l_max,
    sigma=sigma,
    periodic=periodic,
    average=True
)**

# The error occurs during the following line:
**soap_result = soap_descriptor.create(atoms)**

I would greatly appreciate any insights or suggestions on how to resolve this issue with the dscribe library. Perhaps there have been changes to the library, and I need to use an alternative approach or version.

Is there any alternative way to create a SOAP descriptor in dscribe that does not rely on the ext attribute? Any guidance or workaround would be immensely helpful.

Thank you all for your time and expertise. I look forward to your valuable inputs and suggestions.

**Comment by deepakpawar2310** [comment]
This is the SOAP descriptor 

average_soap = SOAP(
    species=["Cu"],
    r_cut=rr_cut,
    n_max=n_max,
    l_max=l_max,
    periodic=False,
    rbf="gto",
    sigma=0.01 * rr_cut
)

**Comment by lauri-codes** [comment]
Hi @deepakpawar2310: Which version of dscribe are you using? You can see this with e.g. `pip list`

**Reply by lauri-codes** [reply]
The `dscribe.ext` module is very important and should be part of every successful installation. I would suspect that something has gone wrong with your installation. In order to debug this, I would need to know your OS, your `dscribe` version, and the way you installed the package (from source, using pip, or using conda).

**Comment by deepakpawar2310** [comment]
Hi
It's working for me !!

However, I am using Ubuntu 220.04 Version !!

Thank you !!


On Wed, Aug 2, 2023 at 9:06 PM Lauri Himanen ***@***.***>
wrote:

> The dscribe.ext module is very important and should be part of every
> successful installation. I would suspect that something has gone wrong with
> your installation. In order to debug this, I would need to know your OS,
> your dscribe version, and the way you installed the package (from source,
> using pip, or using conda).
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/SINGROUP/dscribe/discussions/118#discussioncomment-6618247>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AECR3M4JFTMYIKEZBHKS6ADXTJXYPANCNFSM6AAAAAA2UULIOY>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
*Regards*

Deepak K Pawar
Research Scholar
Department of Mechanical Engineering
IIT Madras, Chennai, India
Mob No: +91-7208385265
